### DISCUSSION TITLE
ISR or SSR not working from next.js app deployed to Google Cloud Run

### DISCUSSION THREAD

**Discussion by ashmore11** [vercel/next.js#19305] [Help]
Hello all!

I'm struggling with an issue where ISR or SSR don't seem to be working on Google Cloud Run. Refreshing the page keeps serving stale content even though I have updated my database.

I thought the issue with ISR might be caused by the instance shutting down after the initial request had completed so next couldn't regenerate in the background. Then I switched to SSR to check and the issue persisted. As long as the instance is alive it will serve stale content.

The same app works perfectly using App Engine flex environment.

Here is my Dockerfile:
```
ARG node_env

FROM node:10

WORKDIR /usr/src/app

ENV NODE_ENV=$node_env

COPY package*.json ./
RUN npm ci --only=production

COPY . .

RUN npm run build

CMD [ "npm", "start" ]
```
I'd appreciate any insights into this : )

**Comment by coderjib** [comment]
Hi, we're seeing the same issue, at least with ISR, did you (or anyone else) manage to find any solution on this? Thanks

**Comment by CiroGomes** [comment]
HI! I was having a similar issue. I use SSR and SSG, but only SSR was working. My project is *deployed* on GCP, using Cloud Run too.

The reason was simple. The instance had to be online for at least the time I put on the ```revalidade```, like this: 
```typescript
export const getStaticProps: GetStaticProps = async () => {
  const hello = 'hello world!';

  return {
    props: {
      hello,
    },
    revalidate: 60,
  };
};
```

On this example, the instance (or pod, as u prefer) should be online for at least 1 min. On my real situation, I had to update the informationns every 4 hours, so the pod never last that longer.

This was my Dockerfile:
```Dockerfile
# base image
FROM node:14-alpine

# Create and change to the app directory.
WORKDIR /usr/app

# Copy application dependency manifests to the container image.
# A wildcard is used to ensure copying both package.json AND package-lock.json (when available).
# Copying this first prevents re-running npm install on every code change.
COPY . .

# Install production dependencies.
# If you add a package-lock.json, speed your build by switching to 'npm ci'.
RUN npm ci

# Copy local code to the container image.

RUN npm run build

# Run the web service on container startup.
CMD [ "npm", "start" ]
```

So the solution was kinda simple. I just followed the docs. You can find the docs here: https://nextjs.org/docs/deployment#docker-image
You will see there a link to their repository, where you can find the Dockerfile, here: https://github.com/vercel/next.js/blob/canary/examples/with-docker/Dockerfile

Just copy and paste the Dockerfile and, if you use .env files, add them on the Dockerfile too, like this: In my case, I had two .env files, the .env.production.local and .env.local, so I added `COPY --from=builder /app/.env.production.local ./` and `COPY --from=builder /app/.env.local ./` right after the others `COPY --from=builder`.

**This also means that I changed from npm to yarn**. On my project, I have no problem changing that, but you may adapt the Dockerfile from the docs to use npm, if you need so.

And that's all! That fixed the issue for me 😆
